src/connectors/NativeConnectorLogic.sol

```diff
@@ -49,7 +49,16 @@ abstract contract NativeConnectorLogic is TokenConnectorLogic {
         if (lockingValue == 0) {
             revert ZeroValueCall();
         }
-        state.addAmount(msg.sender, lockingValue);
-        emit Locked(msg.sender, lockingValue);
+        if (lockingValue < amount) {
+            revert InsufficientFunds(amount, lockingValue);
+        } else if (lockingValue > amount) {
+            (bool success,) = msg.sender.call{value: lockingValue - amount}("");
```

Comment:
Move `state.addAmount(msg.sender, amount)` here before the action happens to be reentrancy-safe.This is not really a problem, but just to be 100% safe